src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,143 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node tail;
+    private Node head;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = new Node(null, value, null);
+            tail = head;
+        } else {
+            head.next = new Node(head, value, null);
+            head = head.next;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Invalid index: " + index + " for size: " + size);
+        }
```

Comment:
move it to the private method

Reply:
It has no sence because I use it once

Reply:
> move it to the private method

It has no sense because we use this once